pkg/fileio/file_io_test.go

```diff
@@ -74,3 +76,76 @@ func TestCopyFile(t *testing.T) {
 		})
 	}
 }
+
+func TestCopyFileN(t *testing.T) {
+	const (
+		destDirPrefix  = "eib-copy-file-n-test-"
+		srcFilePrefix  = "eib-copy-file-n-test-file-"
+		srcFileContent = "CopyFileN test"
+	)
+
+	tmpDir, err := os.MkdirTemp("", destDirPrefix)
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	tmpFile, err := os.CreateTemp(tmpDir, srcFilePrefix)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, tmpFile.Close())
+		require.NoError(t, os.Remove(tmpFile.Name()))
+	}()
+
+	_, err = tmpFile.WriteString(srcFileContent)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name        string
+		source      io.Reader
+		destination string
+		perms       os.FileMode
+		expectedErr string
+	}{
+		{
+			name:        "Destination is an empty file",
+			source:      tmpFile,
+			destination: "",
+			expectedErr: "creating file with permissions: creating file: open : no such file or directory",
+		},
+		{
+			name:        "Destination is a directory",
+			source:      tmpFile,
+			destination: tmpDir,
+			expectedErr: fmt.Sprintf("creating file with permissions: creating file: open %s: is a directory", tmpDir),
+		},
+		{
+			name:        "File is successfully copied",
+			source:      tmpFile,
+			destination: fmt.Sprintf("%s/copy", tmpDir),
+			perms:       NonExecutablePerms,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := CopyFileN(test.source, test.destination, test.perms, 4096)
```

Comment:
Can we also make the chunk size 1 byte only? Using 4096 means that the contents are read in one go.